Packs/Gurucul/Integrations/GuruculGRA/GuruculGRA.py

```diff
@@ -260,50 +271,100 @@ def main() -> None:
             fetch_records(client, '/accounts/orphan', 'Gra.Orphan.Accounts', 'id', params)
 
         elif demisto.command() == 'gra-fetch-resource-orphan-accounts':
-            resource_orphan = '/resources/' + demisto.args().get('resource_name', 'Windows Security') + '/accounts/orphan'
+            resource_name = arguments.get('resource_name', 'Windows Security')
+            resource_orphan = '/resources/' + resource_name + '/accounts/orphan'
             fetch_records(client, resource_orphan, 'Gra.Resource.Orphan.Accounts', 'id', params)
 
         elif demisto.command() == 'gra-user-activities':
-            user_activities_url = '/user/' + demisto.args().get('employee_id') + '/activity'
+            employee_id = arguments.get('employee_id')
+            user_activities_url = '/user/' + employee_id + '/activity'
             fetch_records(client, user_activities_url, 'Gra.User.Activity', 'employee_id', params)
 
         elif demisto.command() == 'gra-fetch-users-details':
-            fetch_records(client, '/users/' + demisto.args().get('employee_id'), 'Gra.User', 'employeeId', params)
+            employee_id = arguments.get('employee_id')
+            fetch_records(client, '/users/' + employee_id, 'Gra.User', 'employeeId', params)
 
         elif demisto.command() == 'gra-highRisk-users':
             fetch_records(client, '/users/highrisk', 'Gra.Highrisk.Users', 'employeeId', params)
 
         elif demisto.command() == 'gra-cases':
-            cases_url = '/cases/' + demisto.args().get('status')
+            status = arguments.get('status')
+            cases_url = '/cases/' + status
             fetch_records(client, cases_url, 'Gra.Cases', 'caseId', params)
 
         elif demisto.command() == 'gra-user-anomalies':
-            anomaly_url = '/users/' + demisto.args().get('employee_id') + '/anomalies/'
+            employee_id = arguments.get('employee_id')
+            anomaly_url = '/users/' + employee_id + '/anomalies/'
             fetch_records(client, anomaly_url, 'Gra.User.Anomalies', 'anomaly_name', params)
 
+        elif demisto.command() == 'gra-case-action':
+            action = arguments.get('action')
+            caseId = arguments.get('caseId')
+            subOption = arguments.get('subOption')
+            caseComment = arguments.get('caseComment')
+            riskAcceptDate = arguments.get('riskAcceptDate')
+            cases_url = '/cases/' + action
+            if action == 'riskManageCase':
+                post_url = '{"caseId":' + caseId + ',"subOption":"' + subOption + '","caseComment":"' + caseComment +\
+                           '","riskAcceptDate":"' + riskAcceptDate + '"}'
+            else:
+                post_url = '{"caseId":' + caseId + ',"subOption":"' + subOption + '","caseComment":"' + \
+                           caseComment + '"}'
+            fetch_post_records(client, cases_url, 'Gra.Case.Action', 'caseId', params, post_url)
```

Comment:
A suggestion for a more readable code:
```suggestion
            post_url = {
            "caseId": caseId,
            "subOption": subOption,
            "caseComment": caseComment,
            }
            if action == 'riskManageCase':
                post_url["riskAcceptDate"] = riskAcceptDate
            
            fetch_post_records(client, cases_url, 'Gra.Case.Action', 'caseId', params, json.dumps(post_url))
```